examples/validation-app/src/__tests__/acceptance/validate.acceptance.ts

```diff
@@ -32,24 +32,63 @@ describe('validate properties', () => {
   });
 
   it('fails when creating a CoffeeShop with the city name exceeds length limit', async () => {
-    const invalidCityNameCS = validCoffeeShop;
+    const invalidCityNameCS = {...validCoffeeShop};
     invalidCityNameCS.city = 'Tooooooooooronto';
-    await client.post('/coffee-shops').send(invalidCityNameCS).expect(422);
+    const response = await client
+      .post('/coffee-shops')
+      .send(invalidCityNameCS)
+      .expect(422);
+    expect(response.body.error.details.length).to.equal(1);
+    expect(response.body.error.details[0].message).to.equal(
+      'City name should be between 5 and 10 characters',
+    );
   });
 
   it('fails when creating a CoffeeShop with an invalid phone number', async () => {
-    const invalidPhoneNumCS = validCoffeeShop;
+    const invalidPhoneNumCS = {...validCoffeeShop};
     invalidPhoneNumCS.phoneNum = '1111111111';
-    await client.post('/coffee-shops').send(invalidPhoneNumCS).expect(422);
+    const response = await client
+      .post('/coffee-shops')
+      .send(invalidPhoneNumCS)
+      .expect(422);
+    expect(response.body.error.details.length).to.equal(1);
+    expect(response.body.error.details[0].message).to.equal(
+      'Invalid phone number',
+    );
   });
 
   it('fails when creating a CoffeeShop with the capacity exceeds limit', async () => {
-    const invalidCapacityCS = validCoffeeShop;
+    const invalidCapacityCS = {...validCoffeeShop};
     invalidCapacityCS.capacity = 10000;
-    await client.post('/coffee-shops').send(invalidCapacityCS).expect(422);
+    const response = await client
+      .post('/coffee-shops')
+      .send(invalidCapacityCS)
+      .expect(422);
+    expect(response.body.error.details.length).to.equal(1);
+    expect(response.body.error.details[0].message).to.equal(
+      'Capacity cannot exceed 100',
+    );
   });
 
-  it('fails when creating a CoffeeShop with an invalid area code', async () => {
+  it('should report all validation errors', async () => {
+    const invalidCapacityTypeCS = {
+      city: '',
+      phoneNum: '',
+      capacity: '',
+      rating: '',
+    };
+    const response = await client
+      .post('/coffee-shops')
+      .send(invalidCapacityTypeCS)
+      .expect(422);
+
+    expect(response.body.error.message).to.equal(
+      'The request body is invalid. See error object `details` property for more info.',
+    );
+    expect(response.body.error.details.length).to.equal(4);
```

Comment:
When this assertion fails on CI, how can we learn more information about the `details` array?

```
Uncaught Error [AssertionError]: expected 3 to be 4
```

At minimum, use something like this:

```suggestion
    expect(response.body.error.details).to.have.length(4);
```

```
Error [AssertionError]: expected Array [ 1, 2, 3 ] to have property length of 4 (got 3)
```

A better solution is to map `details` item into something that will be useful when inspecting test failures. For examples:

```ts
expect(response.body.error.details.map(d => d.code)).to.deepEqual([
  // error code 1
  // error code 2
  // error code 3
  // error code 4
]);
```

This way when the validation fails, we know exactly which validation errors were not reported (or reported erroneously).